sqlglot/optimizer/qualify_columns.py

```diff
@@ -122,40 +125,87 @@ def _expand_using(scope, schema):
                 scope.replace(column, replacement)
 
 
-def _qualify_columns(scope, schema):
-    """Disambiguate columns, ensuring each column specifies a source"""
-    # lazily loaded
-    unambiguous_columns = None
-    all_columns = None
+def _expand_group_by(scope, resolver):
+    group = scope.expression.args.get("group")
+    if not group:
+        return
+
+    # Replace references to select aliases
+    def transform(node, *_):
+        if isinstance(node, exp.Column) and not node.table:
+            table = resolver.get_table(node.name)
+
+            # Source columns get priority over select aliases
+            if table:
+                node.set("table", exp.to_identifier(table))
+                return node
+
+            selects = {s.alias_or_name: s for s in scope.selects}
+
+            select = selects.get(node.name)
+            if select:
+                scope.clear_cache()
+                if isinstance(select, exp.Alias):
+                    select = select.this
+                return select.copy()
+
+        return node
+
+    group = group.transform(transform)
+
+    group.set("expressions", _expand_positional_references(scope, group.expressions))
+    scope.expression.set("group", group)
+
 
+def _expand_order_by(scope):
+    order = scope.expression.args.get("order")
+    if not order:
+        return
+
+    expressions = [o.this for o in order.expressions]
+    for node, new_node in zip(
+        expressions, _expand_positional_references(scope, expressions)
```

Comment:
expressions can just be an inline generator (o.this ...)

Reply:
or just do zip(order.expressions) and then 169 ordered.this.replace

Reply:
I'm not sure I follow completely, let me know if the replacement is OK.

Reply:
i didn't realize you use it twice, what you had was fine, sorry for the swirl